src/renderer/contexts/BackendContext.tsx

```diff
@@ -120,19 +237,39 @@ export const BackendProvider = memo(
             return restartPromiseRef.current;
         }, [backend, refreshNodes]);
 
+        useEffect(() => {
+            if (nodeQueryError && !restartingRef.current) {
+                const message =
+                    nodeQueryError instanceof Error
+                        ? nodeQueryError.message
+                        : String(nodeQueryError);
+                forgetLastErrorAlert();
+                lastErrorAlert.current = sendAlert({
+                    type: AlertType.CRIT_ERROR,
+                    message: `${t(
+                        'error.message.criticalBackend',
+                        'A critical error occurred while processing the node data returned by the backend.'
+                    )}\n\n${t('error.error', 'Error')}: ${message}`,
+                });
+            }
+        }, [nodeQueryError, sendAlert, forgetLastErrorAlert, t]);
+
         const value = useMemoObject<BackendContextState>({
             url,
             backend,
             ownsBackend,
-            schemata,
+            schemata: nodesInfo?.schemata ?? new SchemaMap([]),
             schemaInputs,
             pythonInfo,
-            categories,
-            categoriesMissingNodes,
-            functionDefinitions,
+            categories: nodesInfo?.categories ?? [],
+            categoriesMissingNodes: nodesInfo?.categoriesMissingNodes ?? [],
+            functionDefinitions:
+                nodesInfo?.functionDefinitions ?? new Map<SchemaId, FunctionDefinition>(),
```

Comment:
You disabled memoizing during loading with those empty arrays and maps. Use `EMPTY_ARRAY` and `EMPTY_MAP` instead. For `SchemaMap` we should probably add something like `SchemaMap.EMPTY`.